File: Ex8.java
package Ex8;

import java.util.ArrayList;
import java.util.List;

class Empregado {
    private String nome;
    private String cargo;
    private double salario;

    public Empregado(String nome, String cargo, double salario) {
        this.nome = nome;
        this.cargo = cargo;
        this.salario = salario;
    }

    public String getNome() {
        return nome;
    }

    public String getCargo() {
        return cargo;
    }

    public double getSalario() {
        return salario;
    }

    public void exibirInformacoes() {
        System.out.println("Nome: " + nome + ", Cargo: " + cargo + ", Salário: R$" + salario);
    }
}

class Empresa {
    private String nome;
    private List<Empregado> empregados;

    public Empresa(String nome) {
        this.nome = nome;
        this.empregados = new ArrayList<>();
    }

    public void adicionarEmpregado(Empregado empregado) {
        empregados.add(empregado);
    }

    public void exibirEmpregados() {
        System.out.println("Empregados da empresa " + nome + ":");
        for (Empregado empregado : empregados) {
            empregado.exibirInformacoes();
        }
    }

    public static void main(String[] args) {
        Empresa empresa = new Empresa("Tech Solutions");

        Empregado emp1 = new Empregado("Alice", "Desenvolvedora", 7000.00);
        Empregado emp2 = new Empregado("Bruno", "Analista de Sistemas", 6000.00);
        Empregado emp3 = new Empregado("Carlos", "Gerente de Projetos", 9000.00);

        empresa.adicionarEmpregado(emp1);
        empresa.adicionarEmpregado(emp2);
        empresa.adicionarEmpregado(emp3);

        empresa.exibirEmpregados();
    }
}

File: ex8.go
package main

import (
	"fmt"
)

type Empregado struct {
	Nome  string
	Cargo string
	Salario float64
}

type Empresa struct {
	Nome      string
	Empregados []*Empregado
}

func (e *Empresa) AdicionarEmpregado(emp *Empregado) {
	e.Empregados = append(e.Empregados, emp)
}

func (e Empresa) ExibirEmpregados() {
	fmt.Printf("Empregados da empresa %s:\n", e.Nome)
	for _, empregado := range e.Empregados {
		fmt.Printf("- Nome: %s, Cargo: %s, Salário: R$ %.2f\n", empregado.Nome, empregado.Cargo, empregado.Salario)
	}
}

func main() {
	empresa := Empresa{Nome: "Tech Solutions"}

	empregado1 := &Empregado{Nome: "Alice", Cargo: "Desenvolvedora", Salario: 8000.00}
	empregado2 := &Empregado{Nome: "Bob", Cargo: "Gerente de Projetos", Salario: 12000.00}
	empregado3 := &Empregado{Nome: "Carlos", Cargo: "Analista de Sistemas", Salario: 7000.00}

	empresa.AdicionarEmpregado(empregado1)
	empresa.AdicionarEmpregado(empregado2)
	empresa.AdicionarEmpregado(empregado3)

	empresa.ExibirEmpregados()
}
